### PR TITLE
docs: restructure dev docs

### PR DESCRIPTION
**Pull request by damithc**
Extracted/reordered/rephrased some stuff to improve readability.

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -375,18 +363,64 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g App name, logging level) through the configuration file (default: `config.json`).
 
+== Documentation
+
+We use asciidoc for writing documentation.
+
+[NOTE]
+We chose asciidoc over Markbind because asciidoc, although a bit more complex than Markbind, provides more flexibility in formatting.
```
Comment (pyokagan) [inline docs/DeveloperGuide.adoc]: @damithc Markdown or Markbind? imo, it seems strange to say that Markbind is less complex than asciidoc :-P

Comment (damithc) [reply docs/DeveloperGuide.adoc]: Sorry, should be Markdown. :-p

**Comment by Zhiyuan-Amos** [comment]
Hello prof, same as the other PR, commit 2 are the changes that I've made to your first commit.

**Comment by damithc** [comment]
> Hello prof, same as the other PR, commit 2 are the changes that I've made to your first commit.

Looks good 👍

**Pull request merged**